### PR TITLE
DRY shutdown handlers

### PR DESCRIPTION
**Pull request by Glutexo**
Shutdown handler registration has been extracted to _lib.handlers_. Reused that in the Flask application.

A follow-up to [RHCLOUD-5357](https://projects.engineering.redhat.com/browse/RHCLOUD-5357) #688 and [RHCLOUD-5358](https://projects.engineering.redhat.com/browse/RHCLOUD-5358) #691.

### CODE REVIEW

**Comment by Glutexo** [comment]
Cool, thanks! 👍🏻 We still need one more ✅ to merge. @ryandillinfelton, @shdunning, @jharting

**Comment by Glutexo** [comment]
Maybe @rodrigonull  or @fijshion?

File: app/__init__.py [hunk 1]
```diff
@@ -101,7 +101,7 @@ def create_app(runtime_environment):
 
     db.init_app(flask_app)
 
-    atexit.register(shutdown_hook, db.get_engine(flask_app).dispose, "Database")
+    register_shutdown(db.get_engine(flask_app).dispose, "Closing database")
```
Comment (jharting) [inline app/__init__.py:104]: @Glutexo did you want to remove shutdown_hook function also?

Comment (Glutexo) [reply app/__init__.py:104]: Sure I wanted! Nice catch, thanks!

**Pull request merged**